src/Asset/AssetDefinition.php

```diff
@@ -141,12 +141,31 @@ public static function displayTabContentForItem(CommonGLPI $item, $tabnum = 1, $
     public function showForm($ID, array $options = [])
     {
         $this->initForm($ID, $options);
+        $options['candel'] = false;
+
+        if (!self::isNewID($ID)) {
+            // Add custom delete button that will show the number of assets using this definition and a link to their search page
+            $asset_count = countElementsInTable('glpi_assets_assets', ['assets_assetdefinitions_id' => $ID]);
+            $asset_class = $this->getAssetClassName();
+            $asset_search_url = $asset_class::getSearchURL();
```

Comment:
To make sure that the link will forward to the complete unfiltered list, it should contains the default empty criterai provided by `Glpi\Search\Input\QueryBuilder::getDefaultCriteria()` and the `reset` parameter.